tools/filetop.py

```diff
@@ -20,191 +21,226 @@
-    if (S_ISREG(mode)) {
-        info.type = 'R';
-    } else if (S_ISSOCK(mode)) {
-        info.type = 'S';
-    } else {
-        info.type = 'O';
+    int trace_read_entry(struct pt_regs *ctx, struct file *file,
+        char __user *buf, size_t count)
+    {
+        return do_entry(ctx, file, buf, count, 1);
     }
 
-    struct val_t *valp, zero = {};
-    valp = counts.lookup_or_init(&info, &zero);
-    if (is_read) {
-        valp->reads++;
-        valp->rbytes += count;
-    } else {
-        valp->writes++;
-        valp->wbytes += count;
+    int trace_write_entry(struct pt_regs *ctx, struct file *file,
+        char __user *buf, size_t count)
+    {
+        return do_entry(ctx, file, buf, count, 0);
     }
 
-    return 0;
-}
-
-int trace_read_entry(struct pt_regs *ctx, struct file *file,
-    char __user *buf, size_t count)
-{
-    return do_entry(ctx, file, buf, count, 1);
-}
-
-int trace_write_entry(struct pt_regs *ctx, struct file *file,
-    char __user *buf, size_t count)
-{
-    return do_entry(ctx, file, buf, count, 0);
-}
-
-"""
-if args.tgid:
-    bpf_text = bpf_text.replace('TGID_FILTER', 'tgid != %d' % args.tgid)
-else:
-    bpf_text = bpf_text.replace('TGID_FILTER', '0')
-if args.all_files:
-    bpf_text = bpf_text.replace('TYPE_FILTER', '0')
-else:
-    bpf_text = bpf_text.replace('TYPE_FILTER', '!S_ISREG(mode)')
-
-if debug or args.ebpf:
-    print(bpf_text)
-    if args.ebpf:
-        exit()
-
-# initialize BPF
-b = BPF(text=bpf_text)
-b.attach_kprobe(event="vfs_read", fn_name="trace_read_entry")
-b.attach_kprobe(event="vfs_write", fn_name="trace_write_entry")
-
-DNAME_INLINE_LEN = 32  # linux/dcache.h
-
-print('Tracing... Output every %d secs. Hit Ctrl-C to end' % interval)
-
-# output
-exiting = 0
-while 1:
-    try:
-        sleep(interval)
-    except KeyboardInterrupt:
-        exiting = 1
-
-    # header
-    if clear:
-        call("clear")
-    else:
-        print()
-    with open(loadavg) as stats:
-        print("%-8s loadavg: %s" % (strftime("%H:%M:%S"), stats.read()))
-    print("%-6s %-16s %-6s %-6s %-7s %-7s %1s %s" % ("TID", "COMM",
-        "READS", "WRITES", "R_Kb", "W_Kb", "T", "FILE"))
-
-    # by-TID output
-    counts = b.get_table("counts")
-    line = 0
-    for k, v in reversed(sorted(counts.items(),
-                                key=lambda counts:
-                                  getattr(counts[1], args.sort))):
-        name = k.name.decode()
-        if k.name_len > DNAME_INLINE_LEN:
-            name = name[:-3] + "..."
-
-        # print line
-        print("%-6d %-16s %-6d %-6d %-7d %-7d %1s %s" % (k.pid,
-            k.comm.decode(), v.reads, v.writes, v.rbytes / 1024,
-            v.wbytes / 1024, k.type.decode(), name))
-
-        line += 1
-        if line >= maxrows:
-            break
-    counts.clear()
-
-    countdown -= 1
-    if exiting or countdown == 0:
-        print("Detaching...")
-        exit()
+    """
+
+    def __init__(self, args, debug = 0):
```

Comment:
style: True/False vs 1/0 ?

Reply:
good suggestion, I like a boolean type for clarity